pkg/cmd/server/start/start_master.go

```diff
@@ -306,11 +306,13 @@ func StartMaster(openshiftMasterConfig *configapi.MasterConfig) error {
 
 	unprotectedInstallers := []origin.APIInstaller{}
 
-	authConfig, err := origin.BuildAuthConfig(*openshiftMasterConfig)
-	if err != nil {
-		return err
+	if openshiftMasterConfig.OAuthConfig != nil {
```

Comment:
Added an integration test to make sure cert auth still works in the case where you start up with oauth disabled.